drivers/led/led_pwm.c

```diff
@@ -76,8 +76,8 @@ static int led_pwm_set_brightness(const struct device *dev,
 
 	pulse = led_pwm->period * value / 100;
 
-	return pwm_pin_set_cycles(led_pwm->dev, led_pwm->channel,
-				  led_pwm->period, pulse, led_pwm->flags);
+	return pwm_pin_set_usec(led_pwm->dev, led_pwm->channel,
+				led_pwm->period, pulse, led_pwm->flags);
```

Comment:
I think this won't work for platforms that define the period in devicetree. For example, for: https://github.com/zephyrproject-rtos/zephyr/blob/ab83b6e850b10d03b28be4544040443960841fa1/boards/arm/frdm_k22f/frdm_k22f.dts#L56-L59 where cells are defined as follows: https://github.com/zephyrproject-rtos/zephyr/blob/ab83b6e850b10d03b28be4544040443960841fa1/dts/bindings/pwm/nxp%2Ckinetis-ftm-pwm.yaml#L18-L22 the above code will try to set the period to 15.625 seconds. I doubt it will succeed. And even if it did, the effect would be a really slowly blinking LED.

It looks like rather `pwm_pin_set_nsec()` should be used here and the default value for the `period` field, used when the corresponding property is not present in devicetree, should be changed to 100000.

But maybe there is no need to use here the period value from devicetree and `pwm_pin_set_usec()` with some arbitrary period could be used instead? I don't quite understand why this function was implemented the way it is currently.

Reply:
If I understand correctly we have two ways of controlling Period. We can change function pwm_pin_set_ from cycles to ns or us and we can change value of period parameter. As it is a driver for pwm leds I think that leaving it as it is is not a good idea for reasons already mentioned. Giving some default value for period while using cycles function also isn't a solution because it may not work properly on all devices. I wouldn't like to remove option for setting period value manually. So I will change it as you suggested to nsec and change default value for 100000.